src/engine/filters/enginefilteriir.h

```diff
@@ -62,9 +62,10 @@ class EngineFilterIIR : public EngineFilterIIRBase {
     // this is can be used instead off a final process() call before pause
     // It fades to dry or 0 according to the m_startFromDry parameter
     // it is an alternative for using pauseFillter() calls
-    void processAndPauseFilter(const CSAMPLE* M_RESTRICT pIn,
-            CSAMPLE* M_RESTRICT pOutput,
-            const int iBufferSize) {
+    void processAndPauseFilter(
+            const CSAMPLE* pIn,
+            CSAMPLE* pOutput,
```

Comment:
Should we add a `DEBUG_ASSERT(pIn != pOutput)` for the `m_startFromTry` case?

Reply:
This check doesn't belong here. Instead it could be added to the pIn + pOut functions. But then we need to check that the whole memory ranges don't overlap instead of only if they don't start at the same memory location.